### PR TITLE
Fix Test Coverage Snowflake and Redshift Cluster

### PR DESCRIPTION
**Pull request by bharanidharan14**
Zen hub story id: https://app.zenhub.com/workspaces/airflow-providers-team-61a89ee7fbf048001dc8378d/issues/astronomer/astronomer-providers/87

github issue id: https://github.com/astronomer/astronomer-providers/issues/87

- Fix Snowflake Trigger, Operator, Hooks code coverage above 92
- Fix Redshift Cluster Trigger, Operator code coverage above 96

### CODE REVIEW

File: astronomer/providers/snowflake/hooks/snowflake.py [hunk 1]
```diff
@@ -101,11 +101,9 @@ async def get_query_status(self, query_ids: List[str]):
                                 "type": "FAILED_WITH_ERROR",
                                 "query_id": query_id,
                             }
-                        else:
-                            return status
                     return {"status": "success", "query_ids": sfqid}
             except ProgrammingError as err:
-                error_message = "Programming Error: {0}".format(err)
+                error_message = "Programming Error: {0}".format(str(err))
```
Comment (sunank200) [inline astronomer/providers/snowflake/hooks/snowflake.py:106]: In OSS airflow the convention is to use f string instead of format. The details can be found here: [https://realpython.com/python-f-strings/](https://realpython.com/python-f-strings/)

File: tests/amazon/aws/sensors/test_redshift_sensor.py [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
 def context():
     """
     Creates an empty context.
     """
-    context = {}
-    yield context
+    yield
+
+
+def test_gcs_object_existence_sensor_async():
```
Comment (sunank200) [inline tests/amazon/aws/sensors/test_redshift_sensor.py:25]: Doc string would be useful here.

File: tests/amazon/aws/triggers/test_redshift_cluster.py [hunk 3]
```diff
@@ -37,28 +37,49 @@ def test_redshift_cluster_resume_trigger_serialization():
...
 
-    task = asyncio.create_task(trigger.run().__anext__())
-    await asyncio.sleep(0.5)
-
-    # TriggerEvent was returned
-    assert task.done() is True
-    assert task.result() == TriggerEvent({"status": "success", "cluster_state": "available"})
-    # Prevents error when task is destroyed while in "pending" state
-    asyncio.get_event_loop().stop()
+    # task = asyncio.create_task(trigger.run().__anext__())
```
Comment (sunank200) [inline tests/amazon/aws/triggers/test_redshift_cluster.py:75]: This commented code can be removed.

File: tests/snowflake/hooks/test_snowflake_async.py [hunk 4]
```diff
@@ -90,5 +90,64 @@ async def test_get_query_status(self, mock_conn, query_ids, expected_state, expe
         conn.is_still_running.return_value = False
         conn.get_query_status.return_value = expected_state
         result = await hook.get_query_status(query_ids=query_ids)
-        print("Test case")
         assert result == expected_result
+
```
Comment (sunank200) [inline tests/snowflake/hooks/test_snowflake_async.py:94]: Adding necessary doc strings and comments on the test would be useful in all the tests

**Review by sunank200 — CHANGES_REQUESTED** [review]
There are few changes I have added as comments.

File: astronomer/providers/amazon/aws/triggers/redshift_cluster.py [hunk 5]
```diff
@@ -46,26 +45,28 @@ async def run(self):
...
-        while True:
-            try:
-                if self.operation_type == "resume_cluster":
-                    response = await hook.resume_cluster(cluster_identifier=self.cluster_identifier)
-                    if response:
-                        yield TriggerEvent(response)
-                        return
-                    else:
-                        error_message = f"{self.task_id} failed"
-                        raise AirflowException(error_message)
```
Comment (phanikumv) [inline astronomer/providers/amazon/aws/triggers/redshift_cluster.py:58]: I thought only tests are supposed to be fixed.Why are code changes being done to address code coverage. Did we miss any code during the merge ?

File: astronomer/providers/amazon/aws/triggers/redshift_cluster.py [hunk 6]
```diff
@@ -46,26 +45,28 @@ async def run(self):
...
+                if response:
+                    yield TriggerEvent(response)
+                    return
+                else:
+                    error_message = f"{self.task_id} failed"
+                    yield TriggerEvent({"status": "error", "message": str(error_message)})
+                    return
+        except Exception as e:
+            yield TriggerEvent({"status": "error", "message": str(e)})
+            return
```
Comment (phanikumv) [inline astronomer/providers/amazon/aws/triggers/redshift_cluster.py:69]: I thought only tests are supposed to be fixed.Why are code changes being done to address code coverage. Did we miss any code during the merge ?

File: tests/amazon/aws/operators/test_redshift_cluster.py [hunk 7]
```diff
@@ -84,6 +84,16 @@ def test_resume_cluster_execute_complete(
     mock_log_info.assert_called_with("Resumed cluster successfully, now its in available state")
 
 
+def test_resume_cluster_execute_complete_none():
```
Comment (phanikumv) [inline tests/amazon/aws/operators/test_redshift_cluster.py:87]: may I know what "none" means here in the test name.

File: tests/amazon/aws/operators/test_redshift_cluster.py [hunk 8]
```diff
@@ -145,3 +155,41 @@ def test_pause_cluster_execute_complete(
...
+    task = RedshiftPauseClusterOperatorAsync(
+        task_id="task_test", cluster_identifier="test_cluster", aws_conn_id="aws_conn_test"
+    )
+    with mock.patch.object(task.log, "info") as mock_log_info:
+        task.execute_complete(context=None, event=None)
+    mock_log_info.assert_called_with("%s completed successfully.", "task_test")
+
+
+@mock.patch("airflow.providers.amazon.aws.hooks.redshift_cluster.RedshiftHook.cluster_status")
+def test_pause_cluster_execute_warning(mock_sync_cluster_statue):
```
Comment (phanikumv) [inline tests/amazon/aws/operators/test_redshift_cluster.py:171]: you intended to make the name as mock_sync_cluster_status?

File: tests/amazon/aws/operators/test_redshift_cluster.py [hunk 9]
```diff
@@ -145,3 +155,41 @@ def test_pause_cluster_execute_complete(
...
+
+    with mock.patch.object(redshift_operator.log, "warning") as mock_log_warning:
+        redshift_operator.execute(context=None)
+    mock_log_warning.assert_called_with(
+        "Unable to pause cluster since cluster is currently in status: %s", "paused"
+    )
+
+
+@mock.patch("airflow.providers.amazon.aws.hooks.redshift_cluster.RedshiftHook.cluster_status")
+def test_resume_cluster_execute_warning(mock_sync_cluster_statue):
```
Comment (phanikumv) [inline tests/amazon/aws/operators/test_redshift_cluster.py:185]: you intended to make the name as mock_sync_cluster_status?

File: astronomer/providers/amazon/aws/example_dags/example_redshift_sql.py [hunk 10]
```diff
@@ -1,13 +1,13 @@
-from airflow.models.dag import DAG
-from airflow.utils.timezone import datetime
+import airflow
+from airflow.utils.dates import days_ago
```
Comment (kaxil) [inline astronomer/providers/amazon/aws/example_dags/example_redshift_sql.py:2]: Why this change?

**Comment by bharanidharan14** [comment]
Raised another PR
